src/index.spec.ts

```diff
@@ -1,12 +1,176 @@
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
 import {
   isInteger,
   toLowerCase,
   removeDuplicatesFromArray,
   createRandomProduct,
   getStarWarsPlanets,
-  createProduct 
+  createProduct,
 } from './index';
 
-test('Testing time!', () => {
-  console.log('welcome');
+let noInput: any;
+const emptyString: string = '';
+const emptyArray: any = [];
+const emptyObject: any = {};
+const randomString: any = 'halsfhals';
+const randomNumber: any = 123131;
+const invalidProduct = {
+  name: 'Raw legums',
+  tags: ['vegetable', 'dessert'],
+  description: 'Raw legums on the wooden table',
+  price: 17.11,
+};
+const validProduct = {
+  name: 'Asparagus',
+  tags: ['vegetable'],
+  description: 'Asparagus with ham',
+  price: 18.95,
+};
+
+describe('Takes a number or string and returns a boolean on determining if is integer', () => {
+  test('Should return true', () => {
+    expect(isInteger(8)).toBe(true);
+  });
+  test('Should return false', () => {
+    expect(isInteger('8')).toBe(false);
+  });
+  test('Should return false', () => {
+    expect(isInteger(emptyString)).toBe(false);
+  });
+  test('Should return false', () => {
+    expect(isInteger(noInput)).toBe(false);
+  });
+});
+
+describe('Takes a string and returns it with all characters in lowercase', () => {
+  test('Should return all characters as lowercase', () => {
+    expect(toLowerCase('ABCDEFGHI')).toBe('abcdefghi');
+  });
+  test('Should return an unmodified string', () => {
+    expect(toLowerCase('abcdefghi')).toBe('abcdefghi');
+  });
+  test('Should return characters with lowercase applied when applicable', () => {
+    expect(toLowerCase('I <3 pUnkR0ck!')).toBe('i <3 punkr0ck!');
+  });
+  test('Should return a message', () => {
+    expect(toLowerCase(emptyString)).toBe('Please provide a string');
+  });
+  test('Should return an error', () => {
+    expect(() => {
+      toLowerCase(noInput);
+    }).toThrow;
+  });
+  test('Should return an error', () => {
+    expect(() => {
+      toLowerCase(randomNumber);
+    }).toThrow;
+  });
```

Comment:
You could differentiate these test cases being more descriptive. For example:

```ts
test('Should return an error if no input was passed',()=>{
  // ...
})

test('Should return an error if a number was passed',()=>{
  // ...
})
```

Reply:
Makes total sense. Thanks for the suggestion!